File: lib/src/firebase/cloud_messaging/messages/call_join_request.dart
class CallJoinRequestTokenPayload {
  final String callerUid;
  final String calledUid;
  final String callTransactionId;
  final String callRateStartCents;
  final String callRatePerMinuteCents;
  final String callJoinExpirationTimeUtcMs;

  CallJoinRequestTokenPayload(
      {required this.callerUid, required this.calledUid,
      required this.callTransactionId, required this.callRateStartCents,
      required this.callRatePerMinuteCents,
      required this.callJoinExpirationTimeUtcMs});
}

File: server/call_transaction/src/firebase/fcm/messages/call_join_request.ts
export class CallJoinRequest {
    callerUid: string;
    calledUid: string;

    constructor({callerUid, calledUid}: {callerUid: string, calledUid: string}) {
      this.callerUid = callerUid;
      this.calledUid = calledUid;
    }

    messageType(): string {
      return "call_join_request";
    }
}
